File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.cpp
#include <bits/stdc++.h>

using namespace std;

string solution(string bin1, string bin2) {
    string answer = "";
    int a = 0;
    int b = 0;
    reverse(bin1.begin(), bin1.end());
    reverse(bin2.begin(), bin2.end());
    int value = 1;
    for(int i = 0; i < bin1.size(); i++) {
        a += ((bin1[i] - '0') * value);
        value *= 2;
    }
    value = 1;
    for(int i = 0; i < bin2.size(); i++) {
        b += ((bin2[i] - '0') * value);
        value *= 2;
    }
    int n = a + b;
    if(n == 0) return "0";
    
    while(n > 0) {
        answer = (n % 2 ? '1' : '0') + answer;
        n /= 2;
    }
    return answer;
}

File: 프로그래머스/0/120885. 이진수 더하기/이진수 더하기.swift
import Foundation

/* short code
func solution(_ bin1:String, _ bin2:String) -> String { 
    var a = Int(bin1, radix: 2)!
    var b = Int(bin2, radix: 2)!
    return String(a + b, radix: 2)
}
*/
func addBinary(_ a: String, _ b: String) -> String {
    
    guard let binaryA = Int(a, radix: 2), 
          let binaryB = Int(b, radix: 2) else {
        return "0"
    }
    
    return String(binaryA + binaryB, radix: 2)
}

func solution(_ bin1:String, _ bin2:String) -> String {    
    return addBinary(bin1, bin2)
}